src/state/Logic.ts

```diff
@@ -0,0 +1,51 @@
+import Logic from '../logic/Logic';
+import { Option } from '../permalink/SettingsTypes';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+
+/**
+ * The tracker logic. Logic is static data that is immutable after we've updated it at startup with the options.
+ * This contains either only an error that occured during the initial load, or the successfully
+ * loaded and parsed logic.
+ */
+export interface LogicState {
```

Comment:
Why? I'm happy to adjust this but I'd like to have some rule to follow here.

Reply:
In general I prefer that everything uses the `?` syntax to indicate optionality. As a rule, I'd default to `?` where possible, and only use an explicit `| undefined` if you want to communicate that `undefined` is an explicit value (as opposed to allowing omission), which should generally be pretty rare

Reply:
My general approach is "`| undefined` unless evidence shows a gratuitous number of places where `undefined` is needed in callers / instantiators", because at the end of the day I prefer more explicit `undefined`s over accidentally missing an argument or property.

I'm not too attached to this particular interface - partly because I'd like to make further changes to it later - so I wouldn't lose any sleep over changing this to question marks. I'd argue in this particular case it also makes no real difference since there's little to communicate - this interface exclusively instantiated in this file anyway.